src/sentry/search/utils.py

```diff
@@ -51,74 +51,126 @@ def parse_datetime_range(value):
+        try:
+            return datetime.strptime(value, format).replace(tzinfo=timezone.utc)
+        except ValueError:
+            pass
+
     try:
-        return _parse_datetime_value(value)
-    except (ValueError, IndexError):
-        raise InvalidQuery(u'{} is not a valid datetime query'.format(value))
+        return parse_unix_timestamp(value)
+    except ValueError:
+        pass
 
+    raise InvalidQuery(u'{} is not a valid datetime query'.format(value))
 
-def _parse_datetime_value(value):
-    # this one is fuzzy, and not entirely correct
-    if value.startswith(('-', '+')):
-        return parse_datetime_range(value)
 
+def parse_datetime_comparison(value):
+    if value[:2] == '>=':
+        return (
+            (parse_datetime_string(value[2:]), True),
+            None,
+        )
+    if value[:2] == '<=':
+        return (
+            None,
+            (parse_datetime_string(value[2:]), True),
+        )
+    if value[:1] == '>':
+        return (
+            (parse_datetime_string(value[1:]), False),
+            None,
+        )
+    if value[:1] == '<':
+        return (
+            None,
+            (parse_datetime_string(value[1:]), False),
+        )
+
+    raise InvalidQuery(u'{} is not a valid datetime query'.format(value))
+
+
+def parse_datetime_value(value):
     # timezones are not supported and are assumed UTC
     if value[-1] == 'Z':
         value = value[:-1]
 
-    value_len = len(value)
-    if value_len in (8, 10):
-        value = datetime.strptime(value, '%Y-%m-%d').replace(
+    # A value that only specifies the date (without a time component) should be
+    # expanded to an interval that spans the entire day.
+    if len(value) in (8, 10):
+        value = datetime.strptime(value, DATE_FORMAT).replace(
             tzinfo=timezone.utc,
         )
-        return [value, value + timedelta(days=1)]
-    elif value[4] == '-':
+        return (
+            (value, True),
+            (value + timedelta(days=1), False),
+        )
+
+    # A value that contains the time should converted to an interval.
+    if value[4] == '-':
         try:
-            value = datetime.strptime(value, '%Y-%m-%dT%H:%M:%S').replace(
-                tzinfo=timezone.utc,
-            )
+            value = datetime.strptime(value, DATETIME_FORMAT).replace(tzinfo=timezone.utc)
         except ValueError:
-            value = datetime.strptime(value, '%Y-%m-%dT%H:%M:%S.%f').replace(
-                tzinfo=timezone.utc,
-            )
+            value = datetime.strptime(value, DATETIME_FORMAT_MICROSECONDS).replace(tzinfo=timezone.utc)
     else:
-        value = datetime.utcfromtimestamp(float(value)).replace(
-            tzinfo=timezone.utc,
-        )
-    return [value - timedelta(minutes=5), value + timedelta(minutes=6)]
+        value = parse_unix_timestamp(value)
+
+    return (
+        (value - timedelta(minutes=5), True),
+        (value + timedelta(minutes=6), False),
+    )
```

Comment:
My assumption here is this interval is widened to increase the likelihood of showing results that happened "around that time"?

Reply:
Yeah... this... this one hurts my head a bit.